plugins/modules/purefa_pg.py

```diff
@@ -491,203 +607,328 @@ def update_pgroup(module, array):
+                        snapshot_schedule=SnapshotSchedule(
+                            enabled=module.params["enabled"]
+                        ),
+                        context_names=[module.params["context"]],
+                    ),
                 )
-            except Exception:
+            else:
+                res = array.patch_protection_groups(
+                    group_names=[module.params["name"]],
+                    protection_group=ProtectionGroup(
+                        snapshot_schedule=SnapshotSchedule(
+                            enabled=module.params["enabled"]
+                        )
+                    ),
+                )
+            if res.status_code != 200:
                 module.fail_json(
-                    msg="Changing enabled status of pgroup {0} failed.".format(
-                        module.params["name"]
+                    msg="Changing snapshot enabled state of pgroup {0} failed. Error: {1}".format(
+                        module.params["name"], res.errors[0].message
                     )
                 )
-
     if (
         module.params["volume"]
-        and get_pgroup(module, array)["hosts"] is None
-        and get_pgroup(module, array)["hgroups"] is None
+        and get_pgroup(module, array).host_count == 0
+        and get_pgroup(module, array).host_group_count == 0
     ):
-        if get_pgroup(module, array)["volumes"] is None:
+        if get_pgroup(module, array).volume_count == 0:
             if not module.check_mode:
                 changed = True
-                try:
-                    array.set_pgroup(
-                        module.params["name"], vollist=module.params["volume"]
+                if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):
+                    res = array.post_protection_groups_volumes(
+                        group_names=[module.params["name"]],
+                        member_names=module.params["volume"],
+                        context_names=[module.params["context"]],
                     )
-                except Exception:
+                else:
+                    res = array.post_protection_groups_volumes(
+                        group_names=[module.params["name"]],
+                        member_names=module.params["volume"],
+                    )
+                if res.status_code != 200:
                     module.fail_json(
-                        msg="Adding volumes to pgroup {0} failed.".format(
-                            module.params["name"]
+                        msg="Adding volumes to pgroup {0} failed. Error: {1}".format(
+                            module.params["name"], res.errors[0].message
                         )
                     )
         else:
-            cased_vols = list(module.params["volume"])
-            cased_pgvols = list(get_pgroup(module, array)["volumes"])
+            pgvols = []
+            vols = list(
+                array.get_protection_groups_volumes(
```

Comment:
context case handling is missing, pls add. Its there in multiple occurrences.
When running in fleet array with context, this needs param to be passed - context_names=[module.params["context"]]